### PR TITLE
Replace custom integral types with stdint.h types.

### PR DESCRIPTION
**Pull request by gregcoombe**
The `base/integral_types.h` file defines a number of fixed-width integral types by typedef-ing them to (theoretically) variable-width types. This PR replaces the local types with the common fixed-width types throughout the codebase, and removes this port file.

### CODE REVIEW

**Comment by jmr** [comment]
Can you #include <cstdint> and use std::... instead?

**Comment by gregcoombe** [comment]
Are you suggesting to just rewrite `base/integral_types.h` to use `std::...` types but still continue to include it? My thought was that this simplified the code by removing an unnecessary header. But it does make it a little bit larger diff.

The other reason for using the built-in types is that it reduces the number of custom types that leak into codebases. These types are in the global namespace, not contained within an S2 namespace, so they will interfere with any other types defined elsewhere. Perhaps this is minor concern, since it would strange to have `uint8` defined as anything else except a 8-bit int, but it seemed to me to be worth the effort of removing that potential source of error.

**Comment by jmr** [comment]
> Are you suggesting to just rewrite base/integral_types.h to use std::... types but still continue to include it?

No, just to use `std::int32_t` instead of plain `int32_t`.

**Comment by jmr** [comment]
I will close and re-open this in hope of getting travis-ci to notice the .travis.yml changes. The current CMakeLists.txt is incompatible with the old .travis.yml.

File: src/s2/s1angle_test.cc [hunk 1]
```diff
@@ -71,16 +70,16 @@ TEST(S1Angle, E6E7RepresentationsUnsigned) {
   // Check that unsigned E6/E7 representations work as expected.
   EXPECT_DOUBLE_EQ(
       S1Angle::Degrees(60).radians(),
-      S1Angle::UnsignedE6(static_cast<uint32>(60000000)).radians());
+      S1Angle::UnsignedE6(static_cast<std::uint32_t>(60000000)).radians());
```
Comment (jmr) [inline src/s2/s1angle_test.cc:73]: Let's do the same `using std::uint32_t;` thing in the .cc files that #92 does with `std::string`.

**Comment by jmr** [comment]
Tests are failing.

**Comment by jmr** [comment]
This was done in 755fe37c9fda541b7b9bcb725b03803d2e807dac.

**Pull request closed without merging**